Haptics/app/src/main/java/com/example/android/haptics/samples/ui/views/HomeRoute.kt

```diff
@@ -15,16 +15,199 @@
+                    ) {
+                        for ((index, button) in buttons.withIndex()) {
+                            // Add some spacing before the second button.
+                            if (index == 1) {
+                                Spacer(modifier = Modifier.width(8.dp))
+                            }
+                            HapticFeedbackButton(
+                                label = button.label,
+                                isEnabled = button.worksOnUserDevice,
+                                onClick = {
+                                    if (!button.worksOnUserDevice) {
+                                        coroutineScope.launch {
+                                            onSnackbarMessage("${button.label} is not supported on this device.")
+                                        }
+                                        return@HapticFeedbackButton
+                                    }
+                                    onHapticFeedbackButtonClicked(it, category.categoryType, button.vibrationEffectConstant)
+                                }
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun HapticFeedbackCategory(
+    label: String,
+    content: @Composable () -> Unit
+) {
+    Column {
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.SpaceBetween,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(top = 8.dp, bottom = 16.dp)
+        ) {
+            Text(
+                label,
+                style = MaterialTheme.typography.subtitle2,
+                color = MaterialTheme.colors.subtitleVariant
+            )
+        }
+        content()
+    }
+}
+
+@Composable
+private fun HapticFeedbackButton(
+    label: String,
+    onClick: (view: View) -> Unit,
+    modifier: Modifier = Modifier,
+    isEnabled: Boolean = true,
+) {
+    val alpha = if (isEnabled) 1f else 0.5f
+    val view = LocalView.current
+    TextButton(
+        onClick = { onClick(view) },
+        modifier = modifier
+            .height(64.dp)
+            .width(180.dp)
+            .background(
+                color = MaterialTheme.colors.buttonSurface,
+                shape = MaterialTheme.shapes.large
+            )
+            .alpha(alpha)
+    ) {
+        Text(
+            label,
+            color = MaterialTheme.colors.onButtonSurface,
+            style = MaterialTheme.typography.subtitle2,
+            modifier = Modifier.alpha(alpha)
+        )
+    }
+}
+
+@Preview(showBackground = true)
+@Composable
+fun HomeScreenPreview() {
+    HapticSamplerTheme {
+
+        HomeScreen(
+            homeViewModelState = HomeViewModelState(
+                listOf(
+                    HapticFeedbackCategory(
+                        "Label",
```

Comment:
Maybe this should hold a name closer to the actual one, like "Effects" here and "Primitives" below?
How can I see this preview in the app?

Reply:
Updated string in the preview.

So this preview shows up in Android Studio and really helpful to quickly see the results of your changes without a device. Added a link to explain more about it. But yeah it's just for development similar to what you used to get with XML previews.

https://developer.android.com/jetpack/compose/tooling